ofrak_core/test_ofrak/service/data_service/data_service_interface_test.py

```diff
@@ -375,3 +377,19 @@ async def test_delete_root(self, populated_data_service: DataServiceInterface):
         for data_id in [DATA_0, DATA_1, DATA_2, DATA_3, DATA_4, DATA_5]:
             with pytest.raises(NotFoundError):
                 await populated_data_service.get_by_id(data_id)
+
+    async def test_search_bytes(self, populated_data_service: DataServiceInterface):
+        (results,) = await populated_data_service.search(DATA_0, b"\x00\x10")
+        assert results == 0x10 - 1
+
+        results = await populated_data_service.search(DATA_0, b"\x10", start=0x10)
+        assert results == (0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17)
+
+    async def test_search_regex(self, populated_data_service: DataServiceInterface):
+        results = await populated_data_service.search(DATA_0, re.compile(b"\x00\x10+"))
+        assert results == ((0x10 - 1, b"\x00\x10\x10\x10\x10\x10\x10\x10\x10"),)
+
+        results = await populated_data_service.search(
+            DATA_0, re.compile(b"\x00+\x10+"), start=0xC, end=0x14
+        )
+        assert results == ((0xC, b"\x00\x00\x00\x00\x10\x10\x10\x10"),)
```

Comment:
As you mentioned, should probably add a test for generic string search and a regex string search to be more thorough, otherwise looks good!

Reply:
The method only supports bytes, so a string search is not in scope to test.